### ISSUE TITLE
Conventions for time handling

### ISSUE THREAD

**Issue by kbevers** [OSGeo/PROJ#1242]
I have a confession: I have introduced conflicting conventions for handling time in `helmert` and `deformation`. 

In `helmert` displacements based on rates of changes are calculated as `s = ds*(t_obs - t_central)` whereas in `deformation` the calculation goes `s = ds*(t_central - t_obs)`. The first can be interpreted as "*transform the coordinate FROM the central epoch TO the observation time*" and the other is interpreted the opposite way: "*transform the coordinate FROM the observation time TO the central epoch*". The differing behaviour is documented as such so there isn't a problem per se, but it is somewhat a mental barrier that has to be overcome every time these two operations are used in transformations. I find that I always have to play around with the direction of the operations to get the intended results.

SO, up for discussion is: Should we change the convention for one of the two operations so that they behave similarly?

Personally I prefer to apply the convention used in `helmert` everywhere. It makes the most sense to me to have a central epoch that transformations pivot around instead of the opposite.
 
Changing the convention for `deformation` is obviously not very desirable. However, I feel that it may be better to do it now than to live with conflicting conventions for who knows how long. As far as I know the `deformation` operation hasn't been used in that many cases yet so the ramifications of a change is not likely to be widespread. 

This issue was inspired by #1239.

**Comment by rouault** [comment]
Fortunately, the convention adopted for Helmert conforms to the EPSG Guidance Note.

Regarding `deformation`, I also think that it has probably be adopted by very few people yet. I've no opposition in changing it. Am I right that a consequence of changing the convention would require modifying existing grids to negate their values ? A warning in the documentation would be needed.  (I was wondering if we would need some `multiplier` parameter to be able to switch convention, like done in the vgridshift operation, but this is perhaps unneeded)

**Comment by kbevers** [comment]
The only place I am aware that `deformation` is in use is https://github.com/NordicGeodesy/NordicTransformations/blob/master/resources/NKG which I fortunately have completely control over and will make sure to update accordingly. 

> Am I right that a consequence of changing the convention would require modifying existing grids to negate their values ?

I believe it should be sufficient to change the direction of the operation. Reversing the operation effectively switches `t_obs` and `t_epoch` (central) in `dt = Q->t_epoch - Q->t_obs`. I may be wrong though, it's been a long time since I wrote this code and at this point might as well have been written by someone else :-) I've been trying a few things out just now to see what effect this change would have and I believe I have discovered a (unrelated) bug that I will have to fix first.
